src/usr/local/containerbase/tools/v2/rye.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+function prepare_tool() {
+  local version_codename
+  local tool_path
+
+  version_codename="$(get_distro)"
+  case "${version_codename}" in
+    "focal" | "jammy" | "noble" )
+      apt_install gzip
+        ;;
+    *)
+      echo "Tool '${TOOL_NAME}' not supported on: ${version_codename}! Please use ubuntu 'focal' or 'jammy'." >&2
+      exit 1
+    ;;
+  esac
+
+  create_tool_path > /dev/null
+  export_env RYE_HOME "${USER_HOME}/.rye"
+  export_path "\$RYE_HOME/shims"
+}
+
+function install_tool () {
+  local versioned_tool_path
+  local arch
+  local name
+  local version
+  local platform
+  local repo
+  arch=${ARCHITECTURE}
+  name=${TOOL_NAME}
+  version=${TOOL_VERSION}
+  platform=$(uname -s)
+  repo="astral-sh/rye"
+
+  tool_path=$(find_tool_path)
+
+  if [[ ! -d "${tool_path}" ]]; then
+    if [[ $(is_root) -ne 0 ]]; then
+      echo "${TOOL_NAME} not prepared"
+      exit 1
+    fi
+    prepare_tool
+    tool_path=$(find_tool_path)
+  fi
+
+  if [[ $platform == "Darwin" ]]; then
+    platform="macos"
+  elif [[ $platform == "Linux" ]]; then
+    platform="linux"
+  fi
+
+  if [[ $arch == armv8* ]] || [[ $arch == arm64* ]] || [[ $arch == aarch64* ]]; then
+    arch="aarch64"
+  elif [[ $arch == i686* ]]; then
+    arch="x86"
+  fi
+
+  binary="${name}-${arch}-${platform}"
+  file_name="${binary}.gz"
+  base_url="https://github.com/${repo}/releases/download/${version}"
+  file_url="${base_url}/${file_name}"
+
+  checksum_file=$(get_from_url "${file_url}.sha256")
+  # get checksum from file
+  expected_checksum=$(cut -d' ' -f1  "${checksum_file}")
+
+  file=$(get_from_url "${file_url}" "${file_name}" "${expected_checksum}" "sha256sum")
+
+  versioned_tool_path=$(create_versioned_tool_path)
+  create_folder "${versioned_tool_path}/bin"
+
+  gunzip -c "${file}" > "${versioned_tool_path}/bin/rye"
+  chmod +x "${versioned_tool_path}/bin/rye"
+
+  "${versioned_tool_path}/bin/rye" self install --yes
```

Comment:
what is this doing?

Reply:
rye needs to bootstrap itself. It installs itself to (by default) to `~/.rye` where it stores a copy of itself, after which `<tool_path>/bin/rye` may be removed.
In `~/.rye`, it also stores different versions of python (if needed) and additional tools like `uv` and `ruff` (if needed), as well as a pip registry cache for packages that have been used before.

Reply:
can that folder be configured? we like to support read-only root filesystem, so this is working against it.

can't we simply run it from the installed folder?

Reply:
Yes, it is configurable, so it is possible to install it into the `versioned_tool_path`. It does expect read/write permissions in that dir

Reply:
Rye will now install itself to the versioned tool path